extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheQuery.java

```diff
@@ -40,6 +41,19 @@
      */
     public <T> PanacheQuery<T> project(Class<T> type);
 
+    /**
+     * Generates a SELECT clause with the specified fields.
+     * <p>
+     * The result mapper function is used to build a DTO object from a result object. A result
+     * object is represented either by {@code Object[]} for multiple fields or {@code Object} for a single field.
+     * 
+     * @param <T>
+     * @param fields
+     * @param resultMapper
```

Comment:
Can we call this a `rowMapper` this way it's clear that this is called for every row. I find "result" to be a little bit ambiguous about whether this is called per-row or once for all rows.